data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: clean_query,
+                        clean_consts.DESTINATION_DATASET: date_row.dataset_id,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
+                    })
+            else:
+                # break out of loop to create query, if pid does not exist in table
+                continue
+    print(
+        'Query list complete, retracting ehr deactivated PIDS from the following datasets: %s'
+        % date_columns_df['dataset_id'].tolist())
+    logging.info(
+        'Query list complete, retracting ehr deactivated PIDS from the following datasets: %s'
+        % date_columns_df['dataset_id'].tolist())
+    return queries_list
+
+
+def run_queries(queries):
+    """
+    Function that will perform the retraction.
+
+    :param queries: list of queries to run retraction
+    """
+    query_job_ids = []
+    for query_dict in queries:
+        if 'sandbox' in query_dict['destination_dataset_id']:
+            logging.info('Writing rows to be retracted to, using query %s' %
+                         (query_dict['query']))
+            job_results = bq_utils.query(q=query_dict['query'], batch=True)
+            logging.info('%s table written to %s' %
+                         (query_dict['destination_table_id'],
+                          query_dict['destination_dataset_id']))
+            query_job_id = job_results['jobReference']['jobId']
+            query_job_ids.append(query_job_id)
+        else:
+            logging.info('Truncating table with clean data, using query %s' %
+                         (query_dict['query']))
+            job_results = bq_utils.query(q=query_dict['query'], batch=True)
+            logging.info('%s table updated with clean rows in %s' %
+                         (query_dict['destination_table_id'],
+                          query_dict['destination_dataset_id']))
+            query_job_id = job_results['jobReference']['jobId']
+            query_job_ids.append(query_job_id)
+
+    incomplete_jobs = bq_utils.wait_on_jobs(query_job_ids)
+    if incomplete_jobs:
+        logging.info('Failed on {count} job ids {ids}'.format(
+            count=len(incomplete_jobs), ids=incomplete_jobs))
+        logging.info('Terminating retraction')
+        raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
```

Comment:
Another suggestion, you can do some additional refactoring here.  You can write a function to parse all your arguments and return your parsed arguments.  Then you could have
```
def parse_args(raw_args=None):
    # add your existing parser definition code here
    return parser.parse_args(raw_args)


def main(args=None):
    args = parse_arguments(args)
    query_list = create_queries(args.project_id, args.ticket_number,
                                args.pids_project_id, args.pids_dataset_id,
                                args.pids_table)
    run_queries(query_list)
    logging.info('Retraction complete')

if __name__ == '__main__':
    main()
```

You may want to consider this approach because this will allow the module to be run the same from the command line or imported and used from another python module.

Reply:
thank you for this suggestion-- much cleaner and easier to read. also glad it will make it possible to run the same for cmd line and imported.